controllers/managedfusionoffering_controller.go

```diff
@@ -180,11 +182,33 @@ func (r *ManagedFusionOfferingReconciler) reconcileSubscription() error {
 	return nil
 }
 
+func (r *ManagedFusionOfferingReconciler) addOwnerAnnotation(res client.Object) bool {
+	annotations := res.GetAnnotations()
+	if value, found := annotations[managedFusionAnnotation]; !found ||
+		value != fmt.Sprintf("%s/%s", r.namespace, r.managedFusionOffering.Name) {
+		utils.AddAnnotation(
+			res,
+			managedFusionAnnotation,
+			fmt.Sprintf("%s/%s", r.namespace, r.managedFusionOffering.Name),
+		)
+		return true
+	}
+	return false
```

Comment:
The logic here will be valuable for every annotation addition. Can we move it into the `AddAnnotation` util?
Also, I think this should become a general util function instead of a reconciler function

```
func addOwnerAnnotation(res client.Object, owner client.Object) bool {
	annotations := res.GetAnnotations()
	desiredAnnotationValue := fmt.Sprintf("%s/%s", owner.GetNamespace(), owner.GetName())
	return addAnnotation(res, ownerAnnotation, desiredAnnotationValue)
}
```

Reply:
Moving logic to AddAnnotation won't be good because we have reconcile functions where the object gets updated under CreateOrUpdate function of controller runtime where this logic is already taken care of.

Reply:
But in this case, we will need to pass the owner object which will be managedFusionOffering CR and we are assuming that the offering's reconciler won't have access to it. So I think whatever it is right now is good.
Or we can have a function that will look something like this in utils package
```
func AddOwnerAnnotation(obj metav1.Object, ownerNamespace string, ownerName string, key string) bool {
	annotations := obj.GetAnnotations()
	desiredAnnotationValue := fmt.Sprintf("%s/%s", ownerNamespace, ownerName)
	if value, found := annotations[key]; !found ||
		value != desiredAnnotationValue {
		AddAnnotation(
			obj,
			key,
			desiredAnnotationValue,
		)
		return true
	}
	return false
}
```